activerecord/lib/active_record/fixtures.rb

```diff
@@ -456,6 +456,40 @@ class FixtureClassNotFound < ActiveRecord::ActiveRecordError # :nodoc:
   # In the above example, 'base' will be ignored when creating fixtures.
   # This can be used for common attributes inheriting.
   #
+  # == Composite Primary Key Fixtures
+  #
+  # Fixtures for composite primary key tables are fairly similar to normal tables.
+  # When using an id column, the column may be omitted as usual:
+  #
+  #   # app/models/book.rb
+  #   class Book < ApplicationRecord
+  #     self.primary_key = [:author_id, :id]
+  #   end
+  #
+  #   # books.yml
+  #   alices_adventure_in_wonderland:
+  #     author_id: <%= ActiveRecord::FixtureSet.identify(:lewis_carroll) %>
+  #     title: "Alice's Adventures in Wonderland"
+  #
+  # However, in order to support composite primary key relationships,
+  # you must use the `composite_identify` method:
+  #
+  #   # app/models/book_orders.rb
+  #   class BookOrder < ApplicationRecord
+  #     self.primary_key = [:shop_id, :id]
+  #     belongs_to :order, query_constraints: [:shop_id, :order_id]
+  #     belongs_to :book, query_constraints: [:author_id, :book_id]
+  #   end
+  #
+  #   # book_orders.yml
+  #   alices_adventure_in_wonderland_in_books:
+  #     author_id: <%= ActiveRecord::FixtureSet.identify(:lewis_carroll) %>
+  #     book_id: <%= ActiveRecord::FixtureSet.composite_identify(
+  #                  :alices_adventure_in_wonderland, Book.primary_key)[:id] %>
+  #     order_id: <%= ActiveRecord::FixtureSet.composite_identify(
+  #                  :books, Order.primary_key)[:id] %>
+  #     shop: book_store
```

Comment:
Would we need to use `composite_identify` for `shop_id` here too?

Reply:
Nope, the example implies that the `Shop` model identified by a single column
`composite_identify` could be used but that's unnecessary for non composite PK fixture models

Reply:
Whoops, sorry I meant `identify` -- I'm not sure I follow why `author_id` needs to use `identify` but `shop` can reference the fixture directly? Sorry if I'm missing something obvious! 😅

Reply:
No worries! I think the example implies that in the `author_id` example there is no `belongs_to :author` association on the `Book` model so we can't use `author: lewis_carroll` definition (actually can we? I don't remember if the logic looks for column names or available reflections)

But for the `shop: book_store` we can set it by the association name as the association exists on the `BookOrder` model. And this made me realize that it doesn't exist in the example but I believe it should as I think this is what we wanted to convey 😅

Reply:
I added an author relation with `primary_key: :id`, and used `author: lewis_carroll` just to reduce confusion. I think these non-cpk models can be omitted from the example as they are just simple tables. Otherwise, maybe we should just use an integer to imply they aren't models. Happy to add them if you think it makes more sense.

Reply:
I think the examples you have make sense now! 👍